BUILD

```diff
@@ -0,0 +1,247 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library", "go_test")
+load("@bazel_gazelle//:def.bzl", "gazelle")
+load("@rules_pkg//:rpm.bzl", "pkg_rpm")
+load("@rules_pkg//:pkg.bzl", "pkg_deb", "pkg_tar")
+load("@com_github_bazelbuild_buildtools//buildifier:def.bzl", "buildifier")
+load("@rules_pkg//:mappings.bzl", "pkg_attributes", "pkg_filegroup", "pkg_files", "pkg_mkdirs")
+
+# gazelle:prefix github.com/teslamotors/ansible_puller
+
+gazelle(name = "gazelle")
+
+buildifier(name = "buildifier")
+
+go_library(
+    name = "ansible_puller_lib",
+    srcs = [
+        "ansible.go",
+        "http.go",
+        "http_downloader.go",
+        "idempotent_download.go",
+        "main.go",
+        "s3_downloader.go",
+        "unarchive.go",
+        "util.go",
+        "venv.go",
+    ],
+    importpath = "github.com/teslamotors/ansible_puller",
+    visibility = ["//visibility:private"],
+    deps = [
+        "@com_github_aws_aws_sdk_go_v2//aws",
+        "@com_github_aws_aws_sdk_go_v2_config//:config",
+        "@com_github_aws_aws_sdk_go_v2_feature_s3_manager//:manager",
+        "@com_github_aws_aws_sdk_go_v2_service_s3//:s3",
+        "@com_github_gobuffalo_packr_v2//:packr",
+        "@com_github_gorilla_mux//:mux",
+        "@com_github_pkg_errors//:errors",
+        "@com_github_prometheus_client_golang//prometheus",
+        "@com_github_prometheus_client_golang//prometheus/promhttp",
+        "@com_github_satori_go_uuid//:go_uuid",
+        "@com_github_sirupsen_logrus//:logrus",
+        "@com_github_spf13_pflag//:pflag",
+        "@com_github_spf13_viper//:viper",
+    ],
+)
+
+#
+# Tests
+#
+
+go_test(
+    name = "ansible_test",
+    srcs = ["ansible_test.go"],
+    data = [
+        ":ansible-puller.json",
+        ":testdata",
+    ],
+    embed = [":ansible_puller_lib"],
+    deps = [
+        "@com_github_stretchr_testify//assert",
+        "@com_github_stretchr_testify//suite",
+    ],
+)
+
+go_test(
```

Comment:
I believe this test is redundant because you have the individual tests defined everywhere else.